src/buffer.rs

```diff
@@ -1372,11 +1377,44 @@ impl Buffer {
         self.len == 0
     }
 
-    fn push_str(&mut self, text: &str) {
-        self.ensure(self.len + text.chars().count());
+    fn add_utf8(&mut self, text: &str, item_offset: usize, item_length: usize) {
+        if item_length > (i32::MAX / 8) as usize {
+            return;
+        }
+
+        self.ensure(self.len + item_length * mem::size_of::<u8>() / 4);
```

Comment:
Do we really need size_of for `u8`? It's always 1.

Reply:
Mostly to follow [this part](https://github.com/harfbuzz/harfbuzz/blob/430224b1a06c6eef04de9b6e1ee5aa2cb7a9a403/src/hb-buffer.cc#L1521).

Reply:
I understand, but we don't have to preserve C language limitations.